Nodejs/Product/Nodejs/TypeScriptHelpers/TsConfigJsonFactory.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.NodejsTools.TypeScript
+{
+    internal static class TsConfigJsonFactory
+    {
+        public static async Task<TsConfigJson> CreateAsync(string fullPathToFile)
+        {
+            if (!TypeScriptHelpers.IsTsJsConfigJsonFile(fullPathToFile))
+            {
+                throw new ArgumentException("Expected full path to 'tsconfig.json' file.", nameof(fullPathToFile));
+            }
+
+            if (File.Exists(fullPathToFile))
+            {
+                var retryInterval = 500;
+
+                // populate _source with retries for recoverable errors.
+                for(var attempts = 5; attempts >=0; attempts--)
```

Comment:
It looks like this version might have an extra iteration.  Also, why could backwards?